control-plane/agents/core/src/nexus/states.rs

```diff
@@ -0,0 +1,16 @@
+use crate::core::states::ResourceStates;
+use common_lib::types::v0::store::nexus::NexusState;
+
+/// Implementation of the ResourceSpecs which is retrieved from the ResourceSpecsLocked
+/// During these calls, no other thread can add/remove elements from the list
+impl ResourceStates {
+    /// Get all NexusStates
+    pub async fn get_nexuses(&self) -> Vec<NexusState> {
```

Comment:
get_nexus_state() ?

Reply:
I have renamed the functions for getting the state information so that they are clearer.